Disable translation validation on performance runs.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104887

This PR disables translation validation (TV) when running the benchmark suits on
performance workflows: inductor with A100s.

In summary, the changes are:

- Add flag for turning TV on and off on _benchmarks/dynamo/common.py_
- Turn TV on only on CI accuracy builds
- Add `--no-translation-validation` target flag to _.ci/pytorch/test.sh_

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

ysiraichi committed:
Disable translation validation on performance runs.

This PR disables translation validation (TV) when running the benchmark suits on
performance workflows: inductor with A100s.

In summary, the changes are:

- Add flag for turning TV on and off on _benchmarks/dynamo/common.py_
- Turn TV on only on CI accuracy builds
- Add `--no-translation-validation` target flag to _.ci/pytorch/test.sh_

[ghstack-poisoned]

ysiraichi commented:
Perf run: https://github.com/pytorch/pytorch/actions/runs/5511272974

ezyang commented on .ci/pytorch/test.sh:
```diff
@@ -348,6 +348,8 @@ test_perf_for_dashboard() {
       if [[ "$target" == "performance" ]]; then
         target_flag+=( --cold-start-latency)
+      elif [[ "$target" == "accuracy" ]]; then
+        target_flag+=( --no-translation-validation)
```
we should do this for performance too

ysiraichi replied:
But, we only turn TV on when we pass `--accuracy --ci` arguments. Should I still add the flag for performance?

ysiraichi replied:
@ezyang Is it ok to leave it only for accuracy runs, since we don't turn TV on for performance runs?

ezyang replied:
OK, fair enough.

ysiraichi committed:
Fix error due to py3.8. on "Disable translation validation on performance runs."


This PR disables translation validation (TV) when running the benchmark suits on
performance workflows: inductor with A100s.

In summary, the changes are:

- Add flag for turning TV on and off on _benchmarks/dynamo/common.py_
- Turn TV on only on CI accuracy builds
- Add `--no-translation-validation` target flag to _.ci/pytorch/test.sh_

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy chenyang78

[ghstack-poisoned]

ysiraichi commented on benchmarks/dynamo/common.py:
```diff
@@ -2629,6 +2629,13 @@ def get_example_inputs(self):
         help="Set per-process GPU memory fraction (limit) for reducing usable size and reproducing OOMs",
     )
+
+    parser.add_argument(
+        "--no-translation-validation",
+        action="store_true",
+        help="Disable translation validation for accuracy builds.",
+    )
```
Had to change this in order for it to work with Python 3.8, since, on that version, there's no `argparse.BooleanOptionalAction`.

ysiraichi commented:
[dashboard link](https://hud.pytorch.org/benchmark/compilers?startTime=Tue%2C%2004%20Jul%202023%2012%3A12%3A10%20GMT&stopTime=Tue%2C%2011%20Jul%202023%2012%3A12%3A10%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=gh/ysiraichi/62/orig&lCommit=4c92f9add4cd7aa5d08ba20e9f10b7affb152404&rBranch=main&rCommit=54f33265dba0d0782febe8bf8029efb2d0b29ac1)

I'm not really seeing any compilation time improvement, here. However, I'm pretty sure TV is not running anymore.

Before this PR, we can see in its logs that TV timeout-ed on accuracy runs, when training TIMM model `swin_base_patch4_window7_224` with dynamic shapes. That's not the case for this PR.

ezyang commented:
OK, SGTM

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge
